operator/api/redpanda/v1alpha2/role_types.go

```diff
@@ -81,9 +89,14 @@ type RoleSpec struct {
 	// +kubebuilder:validation:XValidation:message="spec.cluster.staticConfiguration.kafka: required value",rule=`!has(self.staticConfiguration) || has(self.staticConfiguration.kafka)`
 	// +required
 	ClusterSource *ClusterSource `json:"cluster"`
-	// Principals defines the list of users assigned to this role.
-	// Format: Type:Name (e.g., User:john, User:jane). If type is omitted, defaults to User.
+	// Principals lists Redpanda user principals directly assigned to this role.
+	// The operator manages only principals declared here; RoleBinding aggregation
+	// is not implemented yet. Leave empty to manage membership outside the operator.
+	// Format: Type:Name (e.g., User:alice). If type omitted, defaults to User.
+	// Supported types: currently only User (either explicit 'User:' prefix or no prefix).
 	// +kubebuilder:validation:MaxItems=1024
+	// +kubebuilder:validation:items:Pattern=`^(User:.+|[^:]+)$`
+	// +optional
```

Comment:
Same note as below, don't mention RoleBindings.

This is the place to detail how syncing will work. e.g. What happens if there's a pre-existing Role with pre-existing principals? How does the controller decide when a principal should be removed vs left up to user management?